anti-diff/src/Data/FingerTree/RootMeasured/Strict.hs

```diff
@@ -0,0 +1,250 @@
+instance Measured vi a => Measured vi (StrictFingerTree vt vi a) where
+  measure = FT.measure . elements
+
+-- | Re-iteration of @'Measured'@, but for root measures.
+--
+-- This re-iteration is necessary because we want to allow the root measure to
+-- be distinct from the internal measure. For example, we can not create both of
+-- these instances for distinct types @T@ and @T'@:
+--
+-- > @instance Measured T  a where -- ...@
+--
+-- > @instance Measured T' a where -- ...@
+--
+-- Furthermore, we want the root measure to be a @'Group'@ instead of a
+-- @'Monoid'@.
+class Monoid v => RootMeasured v a | a -> v where
+  measureRoot :: a -> v
+
+-- | All @StrictFingerTree@s are root measured.
+instance RootMeasured vt a => RootMeasured vt (StrictFingerTree vt vi a) where
+  measureRoot = tm
+
+-- | Conjunction of @RootMeasured@ and @Measured@ constraints.
+type SuperMeasured vt vi a = (RootMeasured vt a, Measured vi a)
+
+{-------------------------------------------------------------------------------
+  Construction
+-------------------------------------------------------------------------------}
+
+infixl 5 |>
+
+-- | /O(1)/. Add an element to the right end of a sequence.
+--
+-- Mnemonic: a triangle with the single element at the pointy end.
+(|>) ::
+     SuperMeasured vt vi a
+  => StrictFingerTree vt vi a
+  -> a
+  -> StrictFingerTree vt vi a
+SFT vt sft |> (!a) = SFT (vt <> measureRoot a) (sft FT.|> a)
+
+-- | /O(n)/. Create a sequence from a finite list of elements. The opposite
+-- operation 'toList' is supplied by the 'Foldable' instance.
+fromList :: SuperMeasured vt vi a => [a] -> StrictFingerTree vt vi a
+fromList !xs = SFT (foldMap measureRoot xs) (FT.fromList xs)
+
+{-------------------------------------------------------------------------------
+  Splitting
+-------------------------------------------------------------------------------}
+
+-- | /O(?)/. Split a sequence at a point where the predicate on the accumulated
+-- /internal/ measure of the prefix changes from 'False' to 'True'.
+--
+-- For predictable results, one should ensure that there is only one such point,
+-- i.e. that the predicate is /monotonic/.
+--
+-- A function @f@ should be provided that computes the root measures of the left
+-- and right parts of the split. If the @vt@ type has a @'Group'@ instance, then
+-- @f@ is defined for /free/: see the @'splitl'@ and @'splitr'@ variants of the
+-- @'split'@ function.
+--
+-- TODO(jdral): Complexity analysis.
+split ::
+     SuperMeasured vt vi a
+  => (vi -> Bool)
+  -> (vt -> (vt, vt) -> (vt, vt))
+  -> StrictFingerTree vt vi a
+  -> ( StrictFingerTree vt vi a
+     , StrictFingerTree vt vi a
+     )
+split p f (SFT vt sft) = (SFT vtLeft left, SFT vtRight right)
+  where
+    (left, right)     = FT.split p sft
+    (vtLeft, vtRight) = f vt (foldMap measureRoot left, foldMap measureRoot right)
+
+-- | Data type representing either /left/ or /right/.
+data LR = L | R
```

Comment:
Do we need this and `splitlr`? Why not using `splitl` or `splitr` directly?

Reply:
It slightly decrease code duplication in downstream modules liked the `HD.DiffSeq` module in `ouroboros-consensus`. I will look into whether `splitlr` is absolutely necessary, or if there is a better alternative.